src/Proxy.hs

```diff
@@ -372,6 +377,96 @@ runGRPC c k = do
         Nothing -> return yCookieMap
         Just scm -> return $ Map.union scm yCookieMap
 
+runGRPCV2 :: ClientMonad IO (GRPCResultV2 (FromProtoResult a))
+        -> (a -> Handler TypedContent)
+        -> Handler TypedContent
+runGRPCV2 = runGRPCV2WithNotFoundError Nothing
+
+-- |Run a GRPC request.
+runGRPCV2WithNotFoundError ::
+           Maybe ErrorMessage -- ^ Error message to include in the response on GRPC status code NOT_FOUND.
+        -> ClientMonad IO (GRPCResultV2 (FromProtoResult a)) 
+        -> (a -> Handler TypedContent)
+        -> Handler TypedContent
+runGRPCV2WithNotFoundError errM c k = do
+  cfg <- grpcEnvData <$> getYesod
+  cookies <- getCookies
+  $(logOther "Trace") $ "Invoking queries with headers: " <> Text.pack (show cookies)
+  (res, updatedCookies) <- liftIO $ runClientWithCookies cookies cfg c
+  case res of
+    -- A client error occurred.
+    Left clientError -> do
+      $(logError) $ "Internal error accessing GRPC endpoint: " <> Text.pack (show clientError)
+      i <- internationalize
+      sendResponseStatus badGateway502 $ object [
+        "errorMessage" .= i18n i EMGRPCError,
+        "error" .= fromEnum InternalError
+        ]
+    -- Otherwise the HTTP/2 request succeeded and there is a GRPC result.
+    Right r -> do
```

Comment:
So, cfr. the last conversation, I believe this should be scrapped and the result just returned.